src/kaboom.ts

```diff
@@ -527,11 +527,13 @@ function make<T>(comps: CompList<T>): Character<T> {
 
 }
 
-function add<T>(comps: CompList<T>): Character<T> {
-	const obj = make(comps);
+function add<A extends unknown[]>(...comps: [...A]): Character<RecursiveElementType<A>> {
```

Comment:
nit: do you think if this will scare people on autocomplete? is there a way to alias stuff into an easier looking format

Reply:
Yes... But if I'm reading things correctly, as it stands this function's signature is not exposed for autocomplete, only signature exposed is the one in types.ts, for better or worse.

Reply:
Hmm but if it's not exposed in types.ts the new syntax won't resolve in the type check

Reply:
(the `kaboom.d.ts` that's exposed is generated only from `types.ts`)

Reply:
Yeah true. Lemme think about it.